### ISSUE TITLE
Enable parallel builds

### ISSUE THREAD

**Issue by felipehuici** [unikraft/unikraft#6]

**Comment by jdeans289** [comment]
Hi, I'm a student taking a virtualization class at UT Austin, and I'm currently collaborating with one other. @felipehuici  Is it possible that we can get more information about this issue if it's still available?

**Comment by felipehuici** [comment]
Hi Jacqueline, this issue has been taken over by another UT Austin team, I've added an "assigned" label to it just now, sorry this was out of date. If you want you can write us at unikraft@listserv.neclab.eu so we may find another issue, or you can browse through the issue list here.

**Comment by razvand** [comment]
Some clarifications here. The current build system of Unikraft is Makefile-based. When doing parallel builds (i.e. using `make -j <N>`, where `N` is the number of jobs running in parallel), some actions may run **before** after actions that should be dependent. That's because, in a serial build case, prerequisites for a target are processed sequentially. This is not the case in a parallel build, a [common pitfall in Makefile-based parallelization](https://www.cmcrossroads.com/article/pitfalls-and-benefits-gnu-make-parallelization).

New dependency rules need to be added to enforce correctly ordered actions in the case of a parallel build.

**Comment by nderjung** [comment]
@razvand, this potential pitfall has been resolved with #307.

**Comment by razvand** [comment]
@nderjung , there is still an issue that I described in #337

**Comment by nderjung** [comment]
Ah I see, i misunderstood. :)

**Comment by skuenzer** [comment]
Isn't this working now with PR #307?

**Comment by royalvedant** [comment]
Hi @razvand  👋
I came across this issue and I’m interested in solving it.
Is it still open and unassigned?
If yes, I’d like to start working on it.
Thank you for your time!
